### DISCUSSION TITLE
Divided metering

### DISCUSSION THREAD

**Discussion by imranshams** [microsoft/semantic-kernel#7583] [Q&A] [answered]
Hi SK Community,

I'm facing a requirement regarding divided counter by tags or something like that. We are using Azure OpenAI and we want to charge our users based on their Token usage or session based, by default we could find total token usage in Azure AI Studio in metric section.

I was just reading the telemetry doc here:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/docs/TELEMETRY.md

Our approach is creating Kernel in each session start and inject with DI. It will be great if we could assign tags to Kernel in build step and by gathering metrics they will be assigned to counters, so we can track them in billing & metrics in Azure Portal & logs.

Microsoft.SemanticKernel.Connectors.OpenAI - available metrics for OpenAI functionality:
- semantic_kernel.connectors.openai.tokens.prompt (Counter) - number of prompt tokens used.
- semantic_kernel.connectors.openai.tokens.completion (Counter) - number of completion tokens used.
- semantic_kernel.connectors.openai.tokens.total (Counter) - total number of tokens used.

Thank you so much for your time and effort.

**Comment by evchaki** [comment, marked as answer]
@imranshams - we are looking at this.  See issue #7596
